store/search/ts.go

```diff
@@ -38,72 +40,88 @@ type storeImplWithMetrics struct {
 	s Store
 }
 
+func (m *storeImplWithMetrics) measure(ctx context.Context, name string, f func(ctx context.Context) error) {
+	// Low level measurement wrapper that is called by the measure functions on the appropriate receiver
+	tags := metrics.GetSearchTags(ctx, name)
+	if config.DefaultConfig.Metrics.Search.ResponseTime {
+		metrics.SearchRequests.Tagged(tags).Histogram("histogram", tally.DefaultBuckets)
+		defer metrics.SearchRequests.Tagged(tags).Histogram("histogram", tally.DefaultBuckets).Start().Stop()
+	}
+	if config.DefaultConfig.DatadogTrace.Enabled {
```

Comment:
is it possible to move this tracing logic in a single method? We can have a `tracing` package that can have a function startTracing?

```
func startTracing(ctx context.Context, serviceName string, resourceName string, spanType string) (context.Context, func()){
	if config.DefaultConfig.DatadogTrace.Enabled {
		parentSpan, exists := tracer.SpanFromContext(ctx)
		if exists {
			spanOpts := []tracer.StartSpanOption{
				tracer.ServiceName(serviceName),
				tracer.ResourceName(resourceName),
				tracer.SpanType(spanType),
				tracer.Measured(),
			}
			
			// There is a parent span, keep on tracing in the kv layer
			spanOpts = append(spanOpts, tracer.ChildOf(parentSpan.Context()))
			span := tracer.StartSpan(resourceName, spanOpts...)
			defer span.Finish()
			ctx = tracer.ContextWithSpan(ctx, span)
			return ctx, func() {
				span.Finish()
			}
		}
	}

	return ctx, func(){}
}
```

and then the caller can simply do,
```
ctx, cb := StartTracing(...)
defer cb()
```

Reply:
+1 to this as well. It could be a bit distracting to go through this when we aim to debug the behavior of measure. If we encapsulate the tracing logic it will be still visible but the focus in tracing will be... on tracing so it would feel like a more natural scope.

Reply:
Thanks, this is great. I refactored it this way.